cpp/src/parquet/encryption/encryption_internal.cc

```diff
@@ -413,6 +423,12 @@ class AesDecryptor::AesDecryptorImpl {
   }
 
  private:
+  void CheckValid() {
+    if (ctx_ == nullptr) {
```

Comment:
```suggestion
    if (ARROW_PREDICT_FALSE(ctx_ == nullptr)) {
```

Reply:
I was under the impression that compilers automatically treated branches that throw an exception to be unlikely, but I can't find a reference. @felipecrv Do you know about that?

Reply:
Ok, judging from https://www.youtube.com/watch?v=T84swS6DCRo (at 29:00), the gcc compiler already uses such a heuristic.

Reply:
CheckValid should be `const`?

Reply:
Good point @mapleFU